### PR TITLE
Basic maintainer worker

### PR DESCRIPTION
**Pull request by justinvdm**
We need a worker to main the working set of due requests.

### CODE REVIEW

**Comment by justinvdm** [comment]
Depends on #8 and #9, otherwise ready for review.

**Comment by justinvdm** [comment]
Ready for review now.

File: vumi_http_retry/workers/maintainer/worker.py [hunk 1]
```diff
@@ -1,17 +1,65 @@
...
 
     def teardown(self):
-        pass
+        self.redis.transport.loseConnection()
+        self.stop_maintain_loop()
+
+    @inlineCallbacks
+    def maintain(self):
+        prefix = self.config.redis_prefix
+        reqs = yield pop_pending(self.redis, prefix, 0, self.clock.seconds())
```
Comment (hodgestar) [inline vumi_http_retry/workers/maintainer/worker.py:57]: Just flagging that we will need to update this to pop multiple times in a single `maintain` call once `pop_pending` gets a limit argument.

File: vumi_http_retry/workers/maintainer/tests/test_worker.py [hunk 2]
```diff
@@ -1,5 +1,148 @@
...
+
+    @inlineCallbacks
+    def mk_worker(self, config):
+        config['redis_pefix'] = 'test'
+        worker = ToyRetryMaintainerWorker(config)
+
+        yield worker.setup(Clock())
+        self.addCleanup(self.teardown_worker, worker)
+
+        worker.stop_maintain_loop()
```
Comment (hodgestar) [inline vumi_http_retry/workers/maintainer/tests/test_worker.py]: Not quite sure what this `stop_maintain_loop()` call is for here?

Comment (hodgestar) [inline vumi_http_retry/workers/maintainer/tests/test_worker.py]: I would have expected the maintain loop to be running and for the tests to do something like:

```
worker.clock.advance(10)
yield
# check that pending items were moved to ready
```

Comment (hodgestar) [inline vumi_http_retry/workers/maintainer/tests/test_worker.py]: Ah, I see one of the tests restarts the loop explicitly. Maybe `mk_worker` should just have an option for stopping the loop? Then a test either uses a stopped loop or not depending on what it wants to test?

File: vumi_http_retry/workers/maintainer/worker.py (hunk 1 above)
Comment (justinvdm) [inline vumi_http_retry/workers/maintainer/worker.py:57]: Cool, I think that logic should live somewhere the `retries` module and can used from here. Sounds like something that we could potentially want to do from outside of this worker. Could probably be done as part of #12.

File: vumi_http_retry/workers/maintainer/tests/test_worker.py (hunk 2 above)
Comment (justinvdm) [inline vumi_http_retry/workers/maintainer/tests/test_worker.py]: Sounds good.

Comment (justinvdm) [inline vumi_http_retry/workers/maintainer/tests/test_worker.py]: Actually, an option is just the same as calling the method explicitly, will do that instead.

Comment (justinvdm) [inline vumi_http_retry/workers/maintainer/tests/test_worker.py]: Done.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**